validation/src/block_production.rs

```diff
@@ -222,11 +222,11 @@ impl<Client, TxPool, Backend> consensus::Proposer<Block> for Proposer<Client, Tx
 
 			Delay::new(enough_candidates).await;
 
-			tokio_executor::blocking::run(move || {
+			tokio::task::spawn_blocking(move || {
```

Comment:
Does that require a tokio runtime to be running? If so, I'm not a fan.

Reply:
I'm happy to switch to async-std for that.

Reply:
I ran the binary with
```rust
config.tasks_executor = {
    Some(Box::new(move |fut| { async_std::task::spawn(fut); }))
};
``` 
and it worked fine.

Reply:
I think the goal is to get only the light client to compile to WASM. Since this is related to block production I think it's fine that it won't compile. I don't know how @tomaka dealt with the "only compile the light client" part though.

Reply:
I think this specific code isn't compiled when compiling for WASM?

Reply:
Would the best option be to not use `polkadot-validation` as a dependency when compiling on wasm?

Reply:
Yeah, that seems reasonable.

Reply:
If i did that, I'd have to fix this on wasm:
```rust
error[E0432]: unresolved import `polkadot_validation`
  --> network/src/router.rs:27:5
   |
27 | use polkadot_validation::{
   |     ^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `polkadot_validation`

error[E0432]: unresolved import `polkadot_validation`
  --> network/src/validation.rs:23:5
   |
23 | use polkadot_validation::{
   |     ^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `polkadot_validation`

error[E0432]: unresolved import `polkadot_validation`
  --> network/src/validation.rs:52:9
   |
52 | pub use polkadot_validation::Incoming;
   |         ^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `polkadot_validation`

error[E0432]: unresolved import `polkadot_validation`
  --> network/src/gossip.rs:59:5
   |
59 | use polkadot_validation::{SignedStatement};
   |     ^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `polkadot_validation`

error[E0432]: unresolved import `polkadot_validation`
  --> network/src/gossip/attestation.rs:35:5
   |
35 | use polkadot_validation::GenericStatement;
   |     ^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `polkadot_validation`

error[E0433]: failed to resolve: use of undeclared type or module `polkadot_validation`
   --> network/src/gossip.rs:229:14
    |
229 |         let root = polkadot_validation::message_queue_root(
    |                    ^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `polkadot_validation`

error[E0433]: failed to resolve: could not find `polkadot_validation` in `{{root}}`
   --> network/src/gossip.rs:533:6
    |
533 |             ::polkadot_validation::check_statement(
    |               ^^^^^^^^^^^^^^^^^^^ could not find `polkadot_validation` in `{{root}}`

error[E0433]: failed to resolve: use of undeclared type or module `polkadot_validation`
   --> network/src/lib.rs:204:9
    |
204 |         match polkadot_validation::validate_incoming(&self.canon_roots, &pov_block.ingress) {
    |               ^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `polkadot_validation`

error[E0223]: ambiguous associated type
   --> network/src/router.rs:281:61
    |
281 |     fn fetch_pov_block(&self, candidate: &CandidateReceipt) -> Self::FetchValidationProof {
    |                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use fully-qualified syntax: `<router::Router<P, E, T> as Trait>::FetchValidationProof`

error[E0223]: ambiguous associated type
   --> network/src/validation.rs:194:7
    |
194 |     ) -> Self::BuildTableRouter {
    |          ^^^^^^^^^^^^^^^^^^^^^^ help: use fully-qualified syntax: `<validation::ValidationNetwork<P, E, T> as Trait>::BuildTableRouter`

error[E0223]: ambiguous associated type
   --> network/src/validation.rs:244:62
    |
244 |     fn collate(&self, parachain: ParaId, relay_parent: Hash) -> Self::Collation {
    |                                                                 ^^^^^^^^^^^^^^^ help: use fully-qualified syntax: `<validation::ValidationNetwork<P, E, N> as Trait>::Collation`

error: aborting due to 11 previous errors
```